src/Models/SavedToken.php

```diff
@@ -62,4 +70,21 @@ protected function parseAttributes($data)
         }
         return $data;
     }
+
+    /**
+     * Get the path to save the token to.
+     *
+     * @return string The file path for the token file.
+     * @throws \Terminus\Exceptions\TerminusException
+     */
+    protected function getPath()
+    {
+        $path = $this->getConfig()->get('tokens_dir');
+        $id = $this->id;
+        // Reality check to prevent stomping on the local filesystem if there is something wrong with the config.
+        if (!$this->id || !$path) {
+            throw new TerminusException('Could not save the machine token.');
```

Comment:
Yeah. I agree with this, but this is a reality check error. There is no valid reason for this to trigger. It really should just say "Something awful happened!!!". I can make it more specific though. You never know when it might be useful